utils/loading_cache.hh

```diff
@@ -391,11 +401,25 @@ private:
     /// Set the given item as the most recently used item.
     /// The MRU item is going to be at the front of the _lru_list, the LRU item - at the back.
     /// \param lru_entry Cache item that has been "touched"
-    void touch_lru_entry(ts_value_lru_entry& lru_entry) {
+    void touch_lru_entry_2_partitions(ts_value_lru_entry& lru_entry) {
         if (lru_entry.is_linked()) {
-            _lru_list.erase(_lru_list.iterator_to(lru_entry));
+            lru_list_type& lru_list = container_list(lru_entry);
+            lru_list.erase(lru_list.iterator_to(lru_entry));
+        }
+
+        if (lru_entry.touch_count() < PartitionHitThreshold) {
+            _logger.trace("Putting key {} into the new generation partition", lru_entry.key());
+            _new_gen_list.push_front(lru_entry);
+            lru_entry.inc_touch_count();
+        } else {
+            _logger.trace("Putting key {} into the old generation partition", lru_entry.key());
+            _lru_list.push_front(lru_entry);
```

Comment:
Now I'm completely confused :-(
I *thought* (probably wrongly...) that the idea is that highly-"touched" items would be expired based on popularity (LFU), not on recency (LRU). So why are you putting a highly-touched item (touch_count > PartitionHitThreshold) in the LRU list?

Reply:
Because this is a 2 partition cache policy: "new" items live in one partition until they are touched more than `PartitionHitThreshold` times and then they are moved to the other partition.

But both partition's items are organized using LRU list (each of its own) and when we evict from each partition we evict least recently used items first.

Reply:
I see. I didn't understand that this is what you do. I guess it all goes back to me needing to see an explanation of the algorithm (in a commit message, comment, or both) before the actual code so I'll understand what's going on.

Reply:
The algorithm was there from the start, @nyh, in the commit message of the " loading_cache: implement a variation of least frequent recently used (LFRU) eviction policy" patch.